C_implem/linked_lists/linked_lists.c

```diff
@@ -63,36 +63,55 @@ void append_int_list(int val, int_list_t* list) {
     list->length += 1;
 }
 
-void destroy_int_cells(int_cell_t* cell) {
+static void destroy_cells(cell_t* cell) {
     if (cell != NULL) {
-        destroy_int_cells(cell->tail);
+        destroy_cells(cell->tail);
         free(cell);
     }
 }
 
-void destroy_int_list(int_list_t* list) {
-    destroy_int_cells(list->cell);
+void destroy_list(list_t* list) {
+    destroy_cells(list->cell);
     free(list);
 }
 
-void empty_int_list(int_list_t* list) {
-    destroy_int_cells(list->cell);
+void empty_list(list_t* list) {
+    destroy_cells(list->cell);
     list->cell = NULL;
     list->length = 0;
 }
 
-// Visualisation
+// Function with return
+
+void* head_list(list_t* list) {
+    return list->cell->head;
+}
+
+list_t* tail_list(list_t* list) {  //list has to not be empty
```

Comment:
Perhaps you could just return NULL on an empty list?

Reply:
It now returns an error upon receiving an empty list